### ISSUE TITLE
Stale content when editing draft

### ISSUE THREAD

**Issue by alxgrk** [forem/forem#12401]
Hey there,

it seems like the editor's content does not get updated, if a draft you edited once on one machine, is being changed on another machine in the meantime.
I observed that the draft's text is stored in local storage. So the only way to get the updated draft was to clear that entry from local storage.

Maybe the behavior is useful in case you'd otherwise loose your local changes. However, if this is true, then there should probably be a button to update the draft manually.

Best regards

**Comment by cmgorton** [comment]
Hey @alxgrk this looks like it might be related to a few bugs that have been brought up, #4409, #6692, #11753

**Comment by alxgrk** [comment]
Hi @cmgorton, #4409 and #6692 definitely seem related. However, the solution with the clear button from #6692 does not work for me, since I'm using the mobile version, which to my knowledge doesn't have that button. Is this correct?

**Comment by cmgorton** [comment]
Ah good to know @alxgrk. Just for clarification are you using our app or just using the browser on mobile? Mobile on the browser there is a `Revert` button once you add some changes to your post. I am not sure if there is on our App. @fdoxyz do you know if we can revert changes to a post in our App?

**Comment by fdoxyz** [comment]
Hi there, I believe there's currently two things at play here:

1. Re @cmgorton: The revert button does appear visible in the mobile app but the confirmation prompt doesn't show which is definitely a bug for the mobile app (video recording: https://share.getcloudapp.com/wbu9oLEk)
2. @alxgrk are you using the "Basic markdown" or the "Rich + markdown" editor? You can find this in `/settings/customization` (picture below). The problem you mention is something that is indeed limited to local storage per device. If you "Save draft" (or save the article with `published: false`) then you should be able to access the progress on any device

A more powerful editor where changes are in sync across multiple browser tabs/devices in realtime (G docs style) is something that may have been proposed at some point in the past but I'd have to look it up through the open issues


<img width="872" alt="Screen Shot 2021-01-25 at 13 50 31" src="https://user-images.githubusercontent.com/6045239/105757948-4d8cf480-5f14-11eb-84f8-6bf34254e38d.png">

**Comment by alxgrk** [comment]
@cmgorton: I'm using the browser on mobile installed by PWA functionality.

@fdoxyz: I'm using the "Rich + Markdown" editor. And indeed, I clicked "Save draft" on mobile as well as on the computer, but both times I was only able to see the locally changed version. On my laptop I could solve it as described above, but on mobile I don't have the same possibilities. 

I hope, this makes it somehow reproducible? Thanks for your superfast responses btw.

**Comment by fdoxyz** [comment]
It does help to make this reproducible! Thanks again @alxgrk, what I just now did was:

1. Create a draft post in desktop browser
2. Open the post in a mobile browser (a different browser might reproduce this too)
3. Edit the post (add some characters) and save the draft (mobile browser)
4. Edit the draft post again in desktop browser (from step 1)

After doing this I see the older version of the draft appear in the editor despite the draft being saved in the server. I captured this behavior here (check out the characters from the body of the Post):

https://share.getcloudapp.com/mXu1dXWK

Maybe someone from @forem/frontend could take a look whenever they have a chance and introduce an AJAX request to double check the draft is still in sync with local storage. Or if this is already taken in consideration in the codebase there must be a bug somewhere.

On another hand I created an issue to track the "revert" button bug separately
